File: c/claude/p117_nc.c
// C
#include <stdio.h>
#include <stdlib.h>

char* floatToString(double value) {
    char* buffer = malloc(50);
    snprintf(buffer, 50, "%.15g", value);
    return buffer;
}

int main() {
    srand(time(NULL));
    
    double randomFloat1 = (double)rand() / RAND_MAX;
    double randomFloat2 = (double)rand() / RAND_MAX;
    double randomFloat3 = (double)rand() / RAND_MAX;

    char* str_a = floatToString(randomFloat1);
    char* str_b = floatToString(randomFloat2);
    char* str_c = floatToString(randomFloat3);

    free(str_a);
    free(str_b);
    free(str_c);

    return 0;
}

File: java/claude/p117_nc.java
// Java
public class RandomFloatConverter {
    public static String floatToString(double value) {
        return String.valueOf(value);
    }

    public static void main(String[] args) {
        double randomFloat1 = Math.random();
        double randomFloat2 = Math.random();
        double randomFloat3 = Math.random();

        String str_a = floatToString(randomFloat1);
        String str_b = floatToString(randomFloat2);
        String str_c = floatToString(randomFloat3);
    }
}